lib/Myriad/Service/Implementation.pm

```diff
@@ -317,7 +317,16 @@ async method load () {
                     });
                     await $self->rpc->reply_success($service_name, $message, $response);
                 } catch ($e) {
-                    $e = Myriad::Exception::InternalError->new(reason => $e) unless (blessed $e and $e->does('Myriad::Exception'));
+                    # Pass through the myriad exception, use indirect exports from Myriad::Exception::Base
+                    # for simplicity
+                    unless ((ref $e) =~ /^Myriad::Exception::/) {
+                        # Ensure that we are passing the exception reason only by means of scalar
```

Comment:
The comment here is a bit confused: exceptions are _already_ scalars. "Scalar" just means there's only one value, the actual value could be an object, a number, a string, `undef` - anything that isn't a vector.

Reply:
Oops, I meant the non-reference scalars because we expects the exceptions is something like strings but not HASHREFs. Will fix the comment here